### DISCUSSION TITLE
Failed to sync rotation state: additional system role "WindowsDesktop" cannot be applied (not authorized)

### DISCUSSION THREAD

**Discussion by thebigeb** [gravitational/teleport#16231] [Q&A]
I got desktop access working other being able to connect to a Server 2012 R2 server which seems to be unrelated but my logs are filled with these entries

```Sep  8 01:32:18 teleport teleport[361648]: 2022-09-08T01:32:18-04:00 INFO [PROC:1]    Service Instance has updated principals to ["teleport.parity-whats.local"], DNS Names to [], going to request new principals and update. pid:361648.1 service/connect.go:934
Sep  8 01:32:18 teleport teleport[361648]: 2022-09-08T01:32:18-04:00 WARN [PROC:1]    Failed to sync rotation state: additional system role "WindowsDesktop" cannot be applied (not authorized). pid:361648.1 service/connect.go:790
Sep  8 01:32:18 teleport teleport[361648]: 2022-09-08T01:32:18-04:00 WARN [PROC:1]    Sync rotation state cycle failed. Retrying in ~10s pid:361648.1 service/connect.go:687
Sep  8 01:32:27 teleport teleport[361648]: 2022-09-08T01:32:27-04:00 INFO [PROC:1]    Service Instance has updated principals to ["teleport.parity-whats.local"], DNS Names to [], going to request new principals and update. pid:361648.1 service/connect.go:934
Sep  8 01:32:27 teleport teleport[361648]: 2022-09-08T01:32:27-04:00 WARN [PROC:1]    Failed to sync rotation state: additional system role "WindowsDesktop" cannot be applied (not authorized). pid:361648.1 service/connect.go:790
Sep  8 01:32:27 teleport teleport[361648]: 2022-09-08T01:32:27-04:00 WARN [PROC:1]    Sync rotation state cycle failed. Retrying in ~10s pid:361648.1 service/connect.go:687
Sep  8 01:32:36 teleport teleport[361648]: 2022-09-08T01:32:36-04:00 INFO [PROC:1]    Service Instance has updated principals to ["teleport.parity-whats.local"], DNS Names to [], going to request new principals and update. pid:361648.1 service/connect.go:934
Sep  8 01:32:36 teleport teleport[361648]: 2022-09-08T01:32:36-04:00 WARN [PROC:1]    Failed to sync rotation state: additional system role "WindowsDesktop" cannot be applied (not authorized). pid:361648.1 service/connect.go:790
Sep  8 01:32:36 teleport teleport[361648]: 2022-09-08T01:32:36-04:00 WARN [PROC:1]    Sync rotation state cycle failed. Retrying in ~10s pid:361648.1 service/connect.go:687```

**Comment by webvictim** [comment]
Did you upgrade from Teleport v9 to v10 and add another role? I'm wondering whether maybe the join token that you have isn't valid for the `windowsdesktop` role.

**Reply by thebigeb** [reply]
The main teleport server is about 6 weeks old so "I think" its been v10 since the start

**Reply by webvictim** [reply]
What Teleport versions are running on your Teleport server and this `windows_desktop_service` agent?

**Reply by thebigeb** [reply]
Main Server
Teleport v10.2.0 git:v10.2.0-0-gb8cdfe9 go1.18.3
Windows Desktop Agent
Teleport v10.1.9 git:v10.1.9-0-g8f6f6ccbb go1.18.3

**Reply by webvictim** [reply]
I'd try updating your desktop agent to 10.2.0 and see whether the same issue still happens. Technically there is supposed to be full compatibility between minor versions but we have seen bugs with this in the past.

**Reply by thebigeb** [reply]
Updated everything to Teleport v10.2.1 git:v10.2.1-0-g2560469 go1.18.6 and same errors appearing on desktop agaents logs

**Comment by webvictim** [comment]
Run `tctl tokens ls` - does the token you have in your `/etc/teleport.yaml` config file on the Windows agent appear there? What roles are associated with it?

**Reply by thebigeb** [reply]
No active tokens found.

I believe this command is for new invitations? The windows desktop agent is connected and I am able to connect to windows desktops which makes these errors in the logs more confusing
![image](https://user-images.githubusercontent.com/80869534/189810865-30e9a6b2-1734-4e2c-988e-8f478b700a15.png)
